src/lite-yt-embed.js

```diff
@@ -11,37 +11,167 @@
+     *
+     * Worth noting that sddefault is _higher_ resolution than hqdefault. Naming is hard. ;)
+     * From my own testing, it appears that hqdefault is ALWAYS there sddefault is missing for ~10% of videos
+     *
+     * TODO: Do the sddefault->hqdefault fallback
+     *       - When doing this, apply referrerpolicy (https://github.com/ampproject/amphtml/pull/3940)
+     * TODO: Consider using webp if supported, falling back to jpg
+     *
+     * @returns {string}
+     */
+    get posterUrl() {
+        return `https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg`;
+    }
+
+
+    /**
+     * Retrieve the URL targeting the embed video from YouTube.
+     *
+     * @returns {string}
+     */
+    get embedSourceUrl() {
+        return `https://www.youtube-nocookie.com/embed/${this.videoId}?autoplay=1&${this.params}`;
+    }
+
+    static get observedAttributes() {
+        return [
+            'videoid',
+            'params',
+        ];
+    }
+
+    /**
+     * Conduct required actions when an attribute being watched is modified.
+     *
+     * @param {string} name The name of the modified attribute.
+     * @param {string} oldValue The value of the attribute before modification.
+     * @param {string} newValue The value of the attribute after modification.
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+        if (name === 'videoid') {
+            this.videoIdChanged();
+        }
+        if (name === 'params') {
+            this.updateSrc();
+        }
+    }
+
+    /**
+     * Property setter for the video identifier.
+     * This will update the attribute of the HTML to the provided string.
+     *
+     * @param {string} value
+     */
+    set videoId(value) {
+        this.setAttribute('videoid', value);
+    }
+
+    /**
+     * Retrieve the video identifier from the `videoid` attribute.
+     * Encode it for security since it can't necessarily be trusted.
+     * Ref: https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#rule-2---attribute-escape-before-inserting-untrusted-data-into-html-common-attributes
+     *
+     * @returns {string}
+     */
+    get videoId() {
+        return encodeURIComponent(this.getAttribute('videoid'));
+    }
+
+    /**
+     * Set any additional parameters for the video embed.
+     *
+     * @param {string} value
+     */
+    set params(value) {
+        this.setAttribute('params', value);
+    }
+
+    /**
+     * Get any additional parameters for the URL.
+     * Encode them for safety like the video id.
+     * This string will not have an ampersand as the first character.
+     *
+     * @returns {string}
+     */
+    get params() {
+        const parameters = this.getAttribute('params');
+        const hasNoParameters = parameters === null;
+        const encodeKeyAndValue = (param) => {
```

Comment:
for this and the L135 stuff, we should just use `new URLSearchParams()`. it'd clean this up a bit.

Reply:
Oh hey, I forgot that constructor took a string to populate the object with. Nice catch.